addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/CorsScanRule.java

```diff
@@ -0,0 +1,187 @@
+
+                // Evaluates the risk for this alert
+                if (acaoVal.contains("*")) {
+                    risk = Alert.RISK_MEDIUM;
+                } else if (acaoVal.contains(RANDOM_NAME)
+                        || acaoVal.contains("null")
+                        || (secScheme && acaoVal.contains("http:"))) {
+                    // If authenticated AJAX requests are allowed, the risk is higher
+                    risk = (acacVal == "") ? Alert.RISK_MEDIUM : Alert.RISK_HIGH;
+                }
+                Matcher m =
+                        Pattern.compile(
+                                        String.format(
+                                                "^\\s*%s[:\\s]+%s(\\s+%s[:\\s]+%s)?",
+                                                HttpHeader.ACCESS_CONTROL_ALLOW_ORIGIN,
+                                                Pattern.quote(acaoVal),
+                                                ACAC,
+                                                acacVal),
+                                        Pattern.MULTILINE)
+                                .matcher(respHead.toString());
+                buildAlert(risk)
+                        .setMessage(msg)
+                        .setAttack(HttpRequestHeader.ORIGIN + ": " + payload)
+                        .setEvidence(m.find() ? m.group(0) : null)
+                        .raise();
+                return;
+            } catch (IOException e) {
+                LOG.warn(e.getMessage(), e);
+            }
+        }
+    }
+
+    private AlertBuilder buildAlert(int risk) {
+        return newAlert()
+                .setName(risk == Alert.RISK_INFO ? getName() : getConstantStr("vuln.name"))
+                .setRisk(risk)
+                .setConfidence(Alert.CONFIDENCE_HIGH)
+                .setDescription(
+                        risk == Alert.RISK_INFO ? getDescription() : getConstantStr("vuln.desc"));
+    }
+
+    private static String getConstantStr(String suffix) {
+        return Constant.messages.getString("ascanalpha.cors." + suffix);
+    }
+
+    @Override
+    public int getId() {
+        return 40040;
+    }
+
+    @Override
+    public String getName() {
+        return getConstantStr("info.name");
+    }
+
+    @Override
+    public String getDescription() {
+        return getConstantStr("info.desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.SERVER;
+    }
+
+    @Override
+    public String getSolution() {
+        return getConstantStr("soln");
+    }
+
+    @Override
+    public String getReference() {
+        return getConstantStr("refs");
+    }
+
+    @Override
+    public int getCweId() {
+        return 942;
+    }
+
+    @Override
+    public int getWascId() {
+        return 14;
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        List<Alert> alerts = new ArrayList<>();
+        for (int i = Alert.RISK_INFO; i < Alert.RISK_HIGH; i++) {
```

Comment:
Looking at the unit tests there seems to be Info, Med, High alerts. This loop is going to create Info, Low, Med alerts I believe.

Reply:
Yes, the function getExampleAlerts is used for documentation and the Low alert section will be empty. I kept it for consistency with the others scan rules.

Reply:
It should include just the ones that it raises.

Reply:
My point was that the examples should be representative of what the scan rule actually raises.

Reply:
You’ve excluded Low which is great but does the loop condition need to change as well so that High will be included? `<=` bs `>`